src/QMCWaveFunctions/SPOSet.h

```diff
@@ -117,7 +117,9 @@ class SPOSet : public QMCTraits
   // Single Slater creation
   virtual void buildOptVariables(const size_t nel) {}
   // For the MSD case rotations must be created in MultiSlaterDetTableMethod class
-  virtual void buildOptVariables(const std::vector<std::pair<int, int>>& rotations) {}
+  virtual void buildOptVariables(const std::vector<std::pair<int, int>>& rotations,
```

Comment:
In what scenario, rotations and full_rotations are different?
I'd like to get rid of the two buildOptVariables interfaces in SPOSet.h because it makes sense only to RotatedSPO.
Instead, in the current use of this API, we can dynamic_cast and figure out if RotatedSPO is used.
Let us explore changing this in upstream first.

Reply:
The only case where rotations and full_rotations are the same is with 1 electron and 2 orbitals - that leads to 1 parameter and a 2x2 rotation matrix.  I think they are different in all other cases - there are multiple occupied orbitals and/or multiple unoccupied orbitals and the rotations internal to each category are not listed in "rotations", but need to be present in "full_rotations".

Reply:
Oh. Need to document both arguments which are very confusing by just looking at names.